app/src/main/java/org/akvo/flow/event/PermissionAwareLocationListener.java

```diff
@@ -58,7 +58,7 @@ public void stopLocation() {
     }
 
     public void handlePermissionResult(int requestCode, String[] permissions, int[] grantResults) {
-        if (grantResults == null || grantResults.length == 0) {
+        if (grantResults == null || grantResults.length == 0 || permissions.length == 0) {
```

Comment:
* In the first part of the `if` expression, we're considering the case that `grantResults` could be `null`. Could it be the same for `permissions`? 
* In other classes we're annotating parameters with `@Nullable` or `@NonNull`, can we do the same here?

Reply:
good find, I have fixed it